packages/react-native-renderer/src/ReactFiberConfigFabric.js

```diff
@@ -93,8 +96,10 @@ export type Instance = {
     currentProps: Props,
     // Reference to the React handle (the fiber)
     internalInstanceHandle: InternalInstanceHandle,
-    // Exposed through refs.
-    publicInstance: PublicInstance,
+    // Exposed through refs. Potentially lazily created.
+    publicInstance: PublicInstance | null,
+    // This is only necessary to lazily create `publicInstance`.
+    publicRootInstance?: PublicRootInstance | null,
```

Comment:
Is there anyway we could retrieve this lazily from `internalInstanceHandle` for example?

This is an extra slot for every single host component, which we should avoid if we can.

Reply:
Unfortunately `internalInstanceHandle` is pretty much an opaque object here, so we can't tap into the internals. Also, even if we did, the Fiber itself doesn't have a reference to the root, and we'd have to traverse the fiber tree up to it to access this, which would be inefficient.

The benchmarks I used took this field into account already, and even with this it should be a net win.

Reply:
I wouldn't expect this to show on benchmarks, but rather as memory overhead (increased overall heap size).

Reply:
Oh, I only meant runtime performance when I was referencing the benchmarks. What I mean is that this change will overall reduce memory usage, even with that new field, as we'll avoid a large amount of allocations for unused refs.